src/security/forward.h

```diff
@@ -233,5 +247,9 @@ enum {
     SQUID_TLS_ERR_END
 };
 
+#if !USE_OPENSSL
+using X509 = Security::Certificate;
```

Comment:
Is adding an OpenSSL-specific `X509` name to non-OpenSSL builds the best way forward? Would not it be better to change any existing X509 references outside of `USE_OPENSSL` to use Security::Certificate?

Reply:
In principle I can agree with you, but it baloons the scope of this PR way outside its intended goal. This typedef was added to address test failures when neither gnutls nor openssl are used

Reply:
Understood. Given the two remaining options -- keeping some headers non-standalone vs. injecting this bad declaration -- I pick the former.

Reply:
please add an XXX  comment next to it though

Reply:
Thank you. FWIW, adding an XXX would not address my concern.

Reply:
This symbol is no longer needed to pass tests; removing it